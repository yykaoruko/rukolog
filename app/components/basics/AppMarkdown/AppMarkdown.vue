<template>
  <div class="app-markdown">
    <div class="app-markdown__body" v-html="body" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
export default Vue.extend({
  name: 'AppMarkdown',
  props: {
    body: {
      type: String,
      required: true,
    },
  },
});
</script>

<style lang="scss" scoped>
.app-markdown {
  color: inherit;
}
.app-markdown__body {
  margin: 16px auto;
}
.app-markdown__body /deep/ {
  h2,
  h3,
  h4,
  h5 {
    font-size: 1.2rem;
  }
}
</style>
