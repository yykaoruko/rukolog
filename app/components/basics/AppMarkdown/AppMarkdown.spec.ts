import Vue from 'vue';
import { shallowMount, Wrapper } from '@vue/test-utils';
import AppMarkdown from './AppMarkdown.vue';

describe('AppMarkdown', () => {
  describe('mount', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(AppMarkdown, {
        propsData: {
          body: 'body',
        },
      });
    });

    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
