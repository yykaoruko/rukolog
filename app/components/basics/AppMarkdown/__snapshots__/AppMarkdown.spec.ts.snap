// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMarkdown mount snapshot 1`] = `
<div
  class="app-markdown"
>
  <div
    class="app-markdown__body"
  >
    body
  </div>
</div>
`;
